src/test/java/com/example/demo/service/VisitServiceTest.java

```diff
@@ -0,0 +1,183 @@
+
+        VisitDTO result = visitService.bookVisit(request);
+
+        assertEquals(VisitType.SCHEDULED, visit.getVisitType());
+        assertEquals(patient, visit.getPatient());
+
+        assertEquals(1L, result.getVisitId());
+        assertEquals(2L, result.getDoctorId());
+    }
+
+    @Test
+    void getPatientVisits_ValidInput_ShouldReturnPatientVisits() {
+        Long patientId = 1L;
+        Visit visit1 = new Visit();
+        visit1.setId(1L);
+        Patient patient1 = new Patient();
+        patient1.setId(patientId);
+        visit1.setPatient(patient1);
+
+        Visit visit2 = new Visit();
+        visit2.setId(2L);
+
+        List<Visit> allVisits = Arrays.asList(visit1, visit2);
+
+        when(visitRepository.findAll()).thenReturn(allVisits);
+        when(visitMapper.visitToVisitDto(visit1)).thenReturn(new VisitDTO(1L, null, null, null, null, null, null, null));
+        when(visitMapper.visitToVisitDto(visit2)).thenReturn(new VisitDTO(2L, null, null, null, null, null, null, null));
+
+        List<VisitDTO> patientVisits = visitService.getPatientVisits(patientId);
+
+        assertEquals(1, patientVisits.size());
+        assertEquals(1L, patientVisits.get(0).getId());
+    }
+
+    @Test
+    void getDoctorVisits_ValidInput_ShouldReturnDoctorVisits() {
+        Long doctorId = 1L;
+        Doctor doctor = new Doctor();
+        doctor.setId(doctorId);
+        Visit visit1 = new Visit();
+        visit1.setId(1L);
+        visit1.setDoctor(doctor);
+        Doctor doctor1 = new Doctor();
+        doctor1.setId(2L);
+        Visit visit2 = new Visit();
+        visit2.setId(2L);
+        visit2.setDoctor(doctor1);
+
+        List<Visit> doctorVisits = Arrays.asList(visit1, visit2);
+        when(visitRepository.findAll()).thenReturn(doctorVisits);
+
+        VisitDTO visitDTO1 = new VisitDTO(1L, null, null, null, null, null, null, null);
+        VisitDTO visitDTO2 = new VisitDTO(2L, null, null, null, null, null, null, null);
+
+        when(visitMapper.visitToVisitDto(visit1)).thenReturn(visitDTO1);
+        when(visitMapper.visitToVisitDto(visit2)).thenReturn(visitDTO2);
+
+        List<VisitDTO> result = visitService.getDoctorVisits(doctorId);
+
+        assertEquals(1, result.size());
+        assertEquals(1L, result.get(0).getId());
+    }
+
+    @Test
+    void addAvailableVisit_PastDate_ShouldThrowException() {
+        CreateVisitCommand command = new CreateVisitCommand();
+        command.setStartDateTime(LocalDateTime.of(2022, 1, 1, 12, 0)); // Data w przeszłości
+
+        IllegalVisitOperationException exception = assertThrows(IllegalVisitOperationException.class, () -> visitService.addAvailableVisit(command));
+        assertEquals("Cannot create a visit for a past date.", exception.getMessage());
+    }
+
+    @Test
+    void addAvailableVisit_InvalidTime_ShouldThrowException() {
```

Comment:
Jesli to ma byc case ze nie jest do 15 zaokraglona to mozemy nazwac test:
 void addAvailableVisit_TimeNotRoundedTo15_ShouldThrowException()